src/services/submissionService.ts

```diff
@@ -16,20 +16,73 @@ export const getSubmissionById = async (id: string) : Promise<Submission | null>
 
         return convertedEntity
     } catch (error) {
-        throw new Error(`Failed to retrieve submission: ${error}.`)
+        throw new Error(`Error retrieving submission: ${error}`)
     }
 }
 
-function convertToDbSubmission(submission: import('../typeDefs/gqlTypes').Submission): 
-    import('../typeDefs/dbSchema').Submission {
+// TODO add parameters to filter results
+export async function getSubmissions(filters: dbSchema.SubmissionFilters = {}) {
+    try {
+        const db = getFirestore()
+        let subRef: Query<DocumentData> = db.collection('submissions')
+
+        if (filters.googleMapsUrl) {
+            subRef = subRef.where('googleMapsUrl', '==', filters.googleMapsUrl)
+        }
+
+        if (filters.healthcareProfessionalName) {
+            subRef = subRef.where('healthcareProfessionalName', '==', filters.healthcareProfessionalName)
+        }
+
+        if (typeof filters.isUnderReview !== 'undefined') {
+            subRef = subRef.where('isUnderReview', '==', filters.isUnderReview)
+        }
+
+        if (typeof filters.isApproved !== 'undefined') {
+            subRef = subRef.where('isApproved', '==', filters.isApproved)
+        }
+
+        if (typeof filters.isRejected !== 'undefined') {
+            subRef = subRef.where('isRejected', '==', filters.isRejected)
+        }
+
+        const snapshot = await subRef.get()
+
+        let submissions: dbSchema.Submission[] = []
+
+        snapshot.forEach(doc => {
```

Comment:
instead of `.forEach()` + `push()`, you can use the `snapshot.map()` function.

Reply:
Just wanted to add on that the reason .map is be better here is because it returns an array and you could assign it to the variable instead of pushing to it.